packages/langium/src/service-registry.ts

```diff
@@ -89,12 +98,14 @@ export class DefaultServiceRegistry implements ServiceRegistry {
             }
         }
         const ext = UriUtils.extname(uri);
-        const services = this.fileExtensionMap.get(ext);
+        const name = UriUtils.basename(uri);
+        const services = this.fileNameMap.get(name) ?? this.fileExtensionMap.get(ext);
+
         if (!services) {
             if (languageId) {
-                throw new Error(`The service registry contains no services for the extension '${ext}' for language '${languageId}'.`);
+                throw new Error(`The service registry contains no services for the file '${name}' for language '${languageId}'.`);
             } else {
-                throw new Error(`The service registry contains no services for the extension '${ext}'.`);
+                throw new Error(`The service registry contains no services for the file '${name}'.`);
```

Comment:
We should refactor this to retain the existing message when an extension entry is retrieved vs. an entry for a file. Might be confusing otherwise when all messages are changed, especially considering that all projects have just a list of extensions right now.

Reply:
Ok, it was useless to change the message.